### ISSUE TITLE
Hide DS4 no longer working after update

### ISSUE THREAD

**Issue by Exclesa** [Ryochan7/DS4Windows#1472]
I updated the program today to the latest version and everything was working fine most of the day, then suddenly a few minutes ago it stopped working. Saying it could not open it exclusively. I tried restarting multiple times, which seems to make it work fine for about 2 minutes then it just stops working. I tried uninstalling and re-installing the driver from the program and just plain disconnecting and reconnecting my controller. I made sure NOTHING was open that would interfere, Steam, Origin, Nvidia Overlay, etc. Literally all that's open is my browser, and DS4Windows. Nothing that could interfere is enabled on startup either.

Edit: Installed InputMapper HIDGuardian and now it seems to be working. Hopefully it stays that way.

[1111.txt](https://github.com/Ryochan7/DS4Windows/files/5146188/1111.txt)

**Comment by mika-n** [comment]
Yes. The built-in exclusive mode (="Hide DS4 Controller" option in DS4Windows app) is very sensitive, but unfortunately those things are beyond the control of DS4Windows app itself. Sometimes very unexpected apps (fex Edge browse background service even when you are not using the Edge browser) may lock the gamepad and prevent the exclusive mode to work.
https://github.com/Ryochan7/DS4Windows/wik/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

HidGuardian driver from Neferious is more fool proof solution. It is ALTERNATIVE solution, so often there is no need to use both "hide ds4 controller" option and HidGuardian driver at the same time. If you use HidG then untick the "Hide DS4 Controller" option (well, it doesn't hurt to keep it enabled with HidG, but then DS4Windows app just does few unnecessary tricks).

HidGuardian is not exactly InputMapper tool, but a separate app from all mighty Nefarious. DS4Windows docs have instructions on how to install HidG, but in the end the driver is the same even when installed via InputMapper tool.
https://github.com/Ryochan7/DS4Windows/wiki/DS4Windows-and-HIDGuardian-Install-and-Setup-Guide#hidguardian-test-installer

But one word of warning because you seem to have DS4 v1 gamepad. HidGuardian doesn't know the difference between virtual dualshock4 gamepad (created by Nefarious's Vigem driver used by DS4Windows app) and the phystical DS4v1 gamepad. Those both game the same hardware ID. Therefore if you use a profile with dualshock4 output type then it doesn't work with HidG because it hides both the physical and virtual DS4v1 gamepad.
`08/29/2020 21:26:07: Found Controller: 00:01:6C:C5:0C:B8 (USB) (DS4 v.1)`

There is a workaround, but it works only with DS4v2 gamepads (or maybe it works also if you connect the DS4v1 via BT instead of USB). HidNinja tool supplied with DS4Windows can be used to tweak HidGuardian settings and set it to hide only DS4v2 physical controller and leaving DS4v1 intact. This way the virtual DS4v1 gamepad type is left visible for games.
https://github.com/Ryochan7/DS4Windows/wiki/Troubleshooting#games-dont-see-any-controller-not-the-virtual-gamepad-controller-or-physical-controller
https://github.com/Ryochan7/DS4Windows/issues/669#issuecomment-487302990

The built-in exclusiveMode would work with ds4v1 physical gamespads and dualshock4 profiles, but then you just have to try figure out which app or process is locking the gamepad device handle.

**Comment by Ryochan7** [comment]
> Literally all that's open is my browser, and DS4Windows. Nothing that could interfere is enabled on startup either.

It's almost like the HTML5 Gamepad API exists and the Edge browser is one of the biggest culprits for this problem happening. I though Edge was put into that message at some point but it does not look it. Better change that.

If I had the option, I would just remove the **Hide DS4 Controller** option entirely at this point. It just barely works anymore as more software has become greedy about handling HID devices. HidGuardian is the main workaround to the Exclusive mode problem but the vanilla version introduces unacceptable input lag. I have a custom version of HidGuardian that fixes that problem and adds a couple of minor features but I cannot easily redistribute my custom version thanks to Microsoft forcing EV Code Signing on Windows drivers.

Long story short: If you truly want the problem fixed, give me $200 USD so I can properly sign and distribute Windows drivers. I am serious. And don't call me Shirley.

![serious_shirley](https://user-images.githubusercontent.com/581968/91860089-d6875480-ec30-11ea-8a08-9269d0c62de8.gif)

**Comment by mika-n** [comment]
But Edge is mentioned in the Wiki docs page and I suspect this is the case here because these new Win10 versions integrate Edge even deeper in Windows even when the browser is not actively used.
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues
